From ca13fee271ad126832c88896776f3050a6c06e64 Mon Sep 17 00:00:00 2001
From: iglocska <andras.iklody@gmail.com>
Date: Mon, 1 Mar 2021 23:47:01 +0100
Subject: [PATCH] fix: [security] sharing group all org flag too lax

- the all org flag was used as a trigger to make the sharing group obejct itself viewable to all local organisations
- even if the all org flag was set for an instance other than the local one

- as reported by Jeroen Pinoy
---
 app/Model/SharingGroupServer.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/app/Model/SharingGroupServer.php b/app/Model/SharingGroupServer.php
index f96ed5e609..18a7f334f6 100644
--- a/app/Model/SharingGroupServer.php
+++ b/app/Model/SharingGroupServer.php
@@ -98,7 +98,7 @@ public function fetchAllAuthorised()
     public function checkIfAuthorised($id)
     {
         $sg = $this->find('first', array(
-                'conditions' => array('sharing_group_id' => $id, 'all_orgs' => 1),
+                'conditions' => array('sharing_group_id' => $id, 'all_orgs' => 1, 'server_id' => 0),
                 'recursive' => -1,
                 'fields' => array('id'),
         ));
